### PR TITLE
[IMP] migrated event_sale module(remaining files-product.py, sale_order....

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
...py) to new API
- Task: https://www.odoo.com/web#id=12010&view_type=form&model=project.task&action=333&active_id=131
- Pad: https://pad.odoo.com/p/r.Hddf0qvcf9AJY1fL

### CODE REVIEW

File: addons/event_sale/views/sale_order.xml [hunk 1]
```diff
@@ -1,15 +1,13 @@
 <?xml version="1.0"?>
-<openerp>
-    <data>
+<odoo>
         <record model="ir.ui.view" id="event_order_line">
```
Comment (tde-banana-odoo) [inline addons/event_sale/views/sale_order.xml]: You may rename to sale_order_view_form_inherit_event + update name.

File: addons/event_sale/models/sale_order.py [hunk 2]
```diff
@@ -1,53 +1,50 @@
...
-        if redirect_to_event_registration:
-            event_ctx = dict(context, default_sale_order_id=ids[0])
-            return self.pool.get('ir.actions.act_window').for_xml_id(cr, uid, 'event_sale', 'action_sale_order_event_registration', event_ctx)
+    @api.multi
+    def action_button_confirm(self):
+        self.ensure_one()
+        res = super(SaleOrder, self).action_button_confirm()
+        if any(line.event_id for line in self.order_line):
+            return self.env['ir.actions.act_window'].with_context(default_sale_order_id=self.id).for_xml_id('event_sale', 'action_sale_order_event_registration')
         else:
```
Comment (tde-banana-odoo) [inline addons/event_sale/models/sale_order.py]: The else is not necessary.

File: addons/event_sale/views/sale_order.xml (hunk 1 above)
Comment (dja-odoo) [inline addons/event_sale/views/sale_order.xml]: Done!

File: addons/event_sale/models/sale_order.py (hunk 2 above)
Comment (dja-odoo) [inline addons/event_sale/models/sale_order.py]: Done. Removed else part.

File: addons/event_sale/data/event_demo.xml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0"?>
```
Comment (atp-odoo) [inline addons/event_sale/data/event_demo.xml]: add encoding="utf-8"

File: addons/event_sale/models/product.py [hunk 4]
```diff
@@ -1,29 +1,22 @@
...
     _inherit = 'product.product'
-    _columns = {
-        'event_ticket_ids': fields.one2many('event.event.ticket', 'product_id', 'Event Tickets'),
-    }
 
-    def onchange_event_ok(self, cr, uid, ids, type, event_ok, context=None):
-        """ Redirection, inheritance mechanism hides the method on the model """
-        if event_ok:
-            return {'value': {'type': 'service'}}
-        return {}
```
Comment (atp-odoo) [inline addons/event_sale/models/product.py]: why onchange_event_ok removed? it's required .

File: addons/event_sale/views/product.xml [hunk 5]
```diff
@@ -1,13 +1,12 @@
 <?xml version="1.0"?>
```
Comment (atp-odoo) [inline addons/event_sale/views/product.xml]: add encoding="utf-8"

File: addons/event_sale/views/event.xml [hunk 6]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0"?>
```
Comment (atp-odoo) [inline addons/event_sale/views/event.xml]: encoding="utf-8"

File: addons/event_sale/views/sale_order.xml [hunk 7]
```diff
@@ -1,15 +1,13 @@
 <?xml version="1.0"?>
```
Comment (atp-odoo) [inline addons/event_sale/views/sale_order.xml]: encoding="utf-8"

File: addons/event_sale/views/sale_order.xml [hunk 8]
```diff
@@ -1,15 +1,13 @@
...
-<openerp>
-    <data>
-        <record model="ir.ui.view" id="event_order_line">
-             <field name="name">event.sale.order</field>
+<odoo>
+        <record model="ir.ui.view" id="sale_order_view_form_inherit_event">
```
Comment (atp-odoo) [inline addons/event_sale/views/sale_order.xml]: view_sale_order_form_inherit_event

File: addons/event_sale/views/sale_order.xml [hunk 9]
```diff
@@ -1,15 +1,13 @@
...
+             <field name="name">sale.order.view.form.inherit</field>
```
Comment (atp-odoo) [inline addons/event_sale/views/sale_order.xml]: sale.order.form.inherit.event

File: addons/event_sale/data/event_demo.xml (hunk 3 above)
Comment (dja-odoo) [inline addons/event_sale/data/event_demo.xml]: Done.

File: addons/event_sale/models/product.py (hunk 4 above)
Comment (dja-odoo) [inline addons/event_sale/models/product.py]: Is was never being called before. So I removed it.

File: addons/event_sale/views/product.xml (hunk 5 above)
Comment (dja-odoo) [inline addons/event_sale/views/product.xml]: Done.

File: addons/event_sale/views/event.xml (hunk 6 above)
Comment (dja-odoo) [inline addons/event_sale/views/event.xml]: Done.

File: addons/event_sale/views/sale_order.xml (hunk 7 above)
Comment (dja-odoo) [inline addons/event_sale/views/sale_order.xml]: Done.

File: addons/event_sale/views/sale_order.xml (hunk 8 above)
Comment (dja-odoo) [inline addons/event_sale/views/sale_order.xml]: Done.

File: addons/event_sale/views/sale_order.xml (hunk 9 above)
Comment (dja-odoo) [inline addons/event_sale/views/sale_order.xml]: Done.

File: addons/event_sale/models/product.py (hunk 4 above)
Comment (atp-odoo) [inline addons/event_sale/models/product.py]: No it's called. when you create product variant and select event subscription then onchange is triggered and product  become service type.

Comment (dja-odoo) [inline addons/event_sale/models/product.py]: Done.

File: addons/event_sale/views/event.xml [hunk 10]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0"?>
-<openerp>
-    <data>
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
 
         <record model="ir.ui.view" id="view_event_registration_ticket_search">
```
Comment (atp-odoo) [inline addons/event_sale/views/event.xml]: place id before model

File: addons/event_sale/views/event.xml [hunk 11]
```diff
@@ -1,6 +1,5 @@
...
             <field name="name">event.registration.ticket.search</field>
```
Comment (atp-odoo) [inline addons/event_sale/views/event.xml]: name : event.registration.search.inherit

**Comment by atp-odoo** [comment]
1) Place id before model in all views 
2) change view id and name  => "model_name_view_form"  and dot notation my.model.view_type or my.model.view_type.inherit

**Pull request closed without merging**